core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -571,33 +572,51 @@ public void uriCheckGoodSchemeAuthority() throws Exception {
   public void uriCheckNoSchemeAuthority() throws Exception {
     Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
     Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
-    TestAppender appender = new TestAppender();
-    Logger.getRootLogger().addAppender(appender);
-
+    mAppender.track("The URI authority");
+    mAppender.track("The URI scheme");
     AlluxioURI uri = new AlluxioURI("/root");
     mFileSystem.createDirectory(uri);
 
-    if (appender.mEvents.size() > 0) {
-      fail("No events should have been logged.");
-    }
-
-    Logger.getRootLogger().removeAppender(appender);
-    ConfigurationTestUtils.resetConfiguration();
+    assertFalse(mAppender.wasLogged("The URI authority"));
+    assertFalse(mAppender.wasLogged("The URI scheme"));
   }
 
-  private static class TestAppender extends AppenderSkeleton{
+  private static class TestAppender extends AppenderSkeleton {
 
     public List<LoggingEvent> mEvents = new ArrayList<LoggingEvent>();
+    private Hashtable<String, Boolean> mTracked = new Hashtable<>();
```

Comment:
`HashMap` is more useful than `HashTable` in pretty much all scenarios, so we use it exclusively. For the declared type, we usually prefer the shorter `Map<String, Boolean>` over `HashMap<String, Boolean`

https://stackoverflow.com/questions/40471/differences-between-hashmap-and-hashtable

Reply:
I used Hashtable because there was a concurrent access exception when using a HashMap

Reply:
Seemed like there was another thread somewhere doing logging as well